tfjs-backend-webgpu/src/conv2d_mm_vec4_webgpu.ts

```diff
@@ -100,81 +108,45 @@ export class Conv2DMMVec4Program implements WebGPUProgram {
     ];
   }
 
-  // index is used to avoid repeated definition error.
-  getSampleAWithRemainder(index: number): string {
-    return `let flatIndex${
-        index} = getIndexFromCoords4D(coord, uniforms.xShape);
-    let divBy4Remainder${index} = flatIndex${index} % 4;
-    let divBy4Index${index} = flatIndex${index} / 4;
-    let curData${index} = x[divBy4Index${index}];
-    if (divBy4Remainder${index} == 0) {
-      temp = curData${index};
-    } else {
-      // TODO: This could end up being a redundant load with another one in
-      // the same shader invocation. Perhaps there's an opportunity for
-      // optimization
-      let nextData${index} = x[divBy4Index${index} + 1];
-      if (divBy4Remainder${index} == 1) {
-        temp = vec4<f32>(curData${index}.yzw, nextData${index}.x);
-      } else if (divBy4Remainder${index} == 2) {
-        temp = vec4<f32>(curData${index}.zw, nextData${index}.xy);
-      } else if (divBy4Remainder${index} == 3) {
-        temp = vec4<f32>(curData${index}.w, nextData${index}.xyz);
-      }
-    }
-    `;
-  }
-
   getUserCode(): string {
     const matMulSource = makeMatMulPackedVec4Source(
-        this.elementsPerThread, this.tileAOuter, this.tileBOuter,
-        this.tileInner);
-
-    // Below code only applys to valid padding type.
-    const remainderSnippet = this.remainder ?
-        `// The bounds checking is always needed since we use it to pad zero for
-          // the 'same' padding type.
-          if (coordsInBounds4D(coord, uniforms.xShape)) {
-            resData = x[getIndexFromCoords4D(coord, uniforms.xShape) / 4];
-          } else {
-            resData = vec4<f32>(0.0); }` :
-        `var temp = vec4<f32>(0.0);
-          ${this.getSampleAWithRemainder(1)}
-          resData = temp;
-          if (WCol == (uniforms.filterDims[1] - 1)) {
-            coord = vec4<i32>(
-              coord.x, coord.y + 1, coord.z + 1 - uniforms.filterDims[1], 0);
-              ${this.getSampleAWithRemainder(2)}
-            if (inChCoord == 0) {
-              resData = vec4<f32>(resData.xyz, temp.x);
-            } else if (inChCoord == 1) {
-              resData = vec4<f32>(resData.xy, temp.xy);
-            } else {
-              resData = vec4<f32>(resData.x, temp.xyz);
-            }
-          }
-          `;
+        [
+          this.elementsPerThread[0], this.elementsPerThread[1],
+          this.innerElementSize
```

Comment:
Pack these three values for `workPerThread` parameter of makeMatMulPackedVec4Source, and `workPerThread` is defined for output. Is `innerElementSize` related with output, is it reasonable to treat it as a part of `workPerThread`?

Reply:
Split it from workPerThread. See change https://github.com/tensorflow/tfjs/pull/6394/commits/73f4ae4db954ecc28bc6c2e3128b70b54b8b6af5 Thanks.